bundle/src/main/java/com/adobe/acs/commons/mcp/impl/DialogResourceProviderFactoryImpl.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2019 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.mcp.impl;
+
+import com.adobe.acs.commons.mcp.DialogResourceProviderConfiguration;
+import com.adobe.acs.commons.mcp.DialogResourceProviderFactory;
+import com.adobe.acs.commons.mcp.form.DialogProvider;
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.StreamSupport;
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.sling.api.adapter.AdapterFactory;
+import org.apache.sling.spi.resource.provider.ResourceProvider;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+import org.osgi.service.metatype.annotations.Designate;
+import org.slf4j.LoggerFactory;
+
+@Component(
+        service = DialogResourceProviderFactory.class,
+        immediate = true
+)
+@Designate(ocd = DialogResourceProviderConfiguration.class)
+public class DialogResourceProviderFactoryImpl implements DialogResourceProviderFactory {
+
+    private static final org.slf4j.Logger LOG = LoggerFactory.getLogger(DialogResourceProviderFactoryImpl.class);
+    private static final String IMPLEMENTATION_CLASS = "models.adapter.implementationClass";
+    private final Map<String, ServiceRegistration<ResourceProvider>> resourceProviderRegistrations
+            = Collections.synchronizedMap(new HashMap<>());
+    private final Set<String> allKnownModels = Collections.synchronizedSet(new HashSet<>());
+    private final String[] ignoredPackages = new String[]{
```

Comment:
probably not for version 1, but I'd be inclined to make this ignored packages setting part of the `DialogResourceProviderConfiguration`. Would make it a bit more flexible overall.

Reply:
It could but honestly it might not matter too much.  This just helps avoid scanning the hundred or so adapter factories that trigger from product stuff.  For customer bundles there would have to be a lot of additional model packages for additional ignore rules to do anything useful, IMHO.